### PR TITLE
feat: render catchall and the username as rcpt

### LINKED ISSUE #4476: Whitelist is not applied to catchall
**Prior to placing the issue, please check following:** *(fill out each checkbox with an `X` once done)*
- [x] I understand that not following or deleting the below instructions will result in immediate closure and/or deletion of my issue.
- [x] I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [x] I have understood that answers are voluntary and community-driven, and not commercial support.
- [x] I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

## Summary
Catchall aliases are not applied to the rspamd settings map. Accordingly, the classification of spam does not work correctly. For example, the whitelist (-999 spam score) is not applied. Only when writing to the email address or a direct alias.

A wildcard is missing in the settings map:
<sub><sup>(Probably a simple wildcard would do, but I haven't looked into the rspamd configuration in detail)<sub><sup>

```
rcpt = "/^*[+].*@domain.tld$/i";
rcpt = "*@domain.tld";
```

image
image_2022-02-18_12-01-37

## Logs
--

## Reproduction
1. Login as Mailbox User to Mailcow.
2. Go to Tab `Spam Filter`
3. Add a sender address to the whitelist
4. Send from this address directly to the mailbox
5. Send from this address to an catchall-alias of this address
6. See that the whitelist is applied to the direct email (-1000 spam score) and not to the catchall-alias

## System information
| Question | Answer |
| --- | --- |
| My operating system | Ubuntu 20.04.3 LTS |
| Is Apparmor, SELinux or similar active? | nope |
| Virtualization technlogy (KVM, VMware, Xen, etc - **LXC and OpenVZ are not supported** | - |
| Server/VM specifications (Memory, CPU Cores) | - |
| Docker Version (`docker version`) | Docker Engine - Community 20.10.11 |
| Docker-Compose Version (`docker-compose version`) | docker-compose version 1.29.2, build 5becea4c |
| Reverse proxy (custom solution) | traefik |

- Output of `git diff origin/master`, any other changes to the code? -> nope

### PR DESCRIPTION
**Pull request by fionera**
This fixes #4476

### CODE REVIEW

**Comment by DerLinkman** [comment]
Thanks, was looking into it as well.

Did you test everything?

**Comment by RafaelKr** [comment]
Were the line endings changed or why does the diff show "identical" lines as changed?

Edit: Okay, was just fixed :D

**Comment by fionera** [comment]
Yeah the line endings where broken. IntelliJ was kinda freaked out :D I checked the output of the script and it should work.

```
  whitelist_fionerafionerade {
    from = "/^.*@lists\.php\.net$/i";
    priority = 6;
    rcpt = "/^fionera[+].*@fionera.de$/i";
    rcpt = "fionera@fionera.de";
    rcpt = "/^*[+].*fionera.de$/i";
    apply "default" {
      MAILCOW_WHITE = -999.0;
    }
    symbols [
      "MAILCOW_WHITE"
    ]
  }
```

Edit: I just saw that I am missing *@fionera.de. Let me fix that ^^

**Comment by fionera** [comment]
Ok turns out: rspamd supports @domain.com. I just replaced all that logic with one line ^^'
https://www.rspamd.com/doc/configuration/settings.html

```
  whitelist_fionerafionerade {
    from = "/^.*@lists\.php\.net$/i";
    priority = 6;
    rcpt = "/^fionera[+].*@fionera.de$/i";
    rcpt = "fionera@fionera.de";
    rcpt = "@fionera.de";
    apply "default" {
      MAILCOW_WHITE = -999.0;
    }
    symbols [
      "MAILCOW_WHITE"
    ]
  }
  ```

Yes the other entries for that domain are basically useless but I dont know if I should clean it up that far

**Comment by andryyy** [comment]
A catch all? So you basically controll the whitelist for all users of a domain because you hold the catch all? 

That’s what a domain whitelist is for.

File: data/conf/rspamd/dynmaps/settings.php [hunk 1]
```diff
@@ -97,15 +97,27 @@ function ucl_rcpts($object, $type) {
   global $pdo;
   $rcpt = array();
   if ($type == 'mailbox') {
+    // The mailbox itself
+    {
```
Comment (andryyy) [inline data/conf/rspamd/dynmaps/settings.php]: Why putting it in { } ?

Comment (fionera) [reply data/conf/rspamd/dynmaps/settings.php]: I didnt want to pollute the scope but just noticed that PHP doesnt support that. I will remove them

**Comment by fionera** [comment]
> A catch all? So you basically controll the whitelist for all users of a domain because you hold the catch all?
> 
> That’s what a domain whitelist is for.

I would handle it that way, since you are receiving all Emails anyway.

**Comment by milkmaker** [comment]
This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

**Comment by fionera** [comment]
Is there any progress with this or do I have to keep my patch locally?

**Comment by milkmaker** [comment]
This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

**Comment by milkmaker** [comment]
This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

**Pull request closed without merging**